addons/calendar/models/calendar.py

```diff
@@ -707,6 +707,10 @@ def _compute_is_highlighted(self):
     privacy = fields.Selection([('public', 'Everyone'), ('private', 'Only me'), ('confidential', 'Only internal users')], 'Privacy', default='public', states={'done': [('readonly', True)]}, oldname="class")
     location = fields.Char('Location', states={'done': [('readonly', True)]}, track_visibility='onchange', help="Location of Event")
     show_as = fields.Selection([('free', 'Free'), ('busy', 'Busy')], 'Show Time as', states={'done': [('readonly', True)]}, default='busy')
+    activity_id = fields.Many2one('mail.activity', string='Mail Activity')
+    res_id = fields.Integer('Related Document ID', index=True)
+    res_model_id = fields.Many2one('ir.model', 'Related Document Model', index=True, ondelete='cascade')
+    res_model = fields.Char('Related Document Model', index=True, related='res_model_id.model', store=True)
```

Comment:
After discussing it with @jem-odoo it's good to store them indeed.
We should use this mechanism in crm and hr_applicant to replace the `applicant_id` and `opportunity_id` fields. It will also be nice to have the related document button for them too.
We can remove the indexes though, they shouldn't be needed.